trestle/core/resolver/modify.py

```diff
@@ -39,59 +39,75 @@ def __init__(
         profile: prof.Profile,
         change_prose: bool = False,
         block_adds: bool = False,
-        params_format: str = None
+        params_format: str = None,
+        param_rep: ParameterRep = ParameterRep.VALUE_OR_LABEL_OR_CHOICES
     ) -> None:
         """Initialize the filter."""
         self._profile = profile
         self._catalog_interface: Optional[CatalogInterface] = None
         self._block_adds = block_adds
         self._change_prose = change_prose
         self._params_format = params_format
+        self._param_rep = param_rep
         logger.debug(f'modify initialize filter with profile {profile.metadata.title}')
 
     @staticmethod
-    def _replace_id_with_text(prose: str, param_dict: Dict[str, str]) -> str:
-        """Find all instances of param_id in prose and replace with param_text.
+    def _replace_param_id_with_str(prose: str, param_id: str, param_str: str) -> str:
+        """Replace param_id's with param_str and check for param_1 vs. param_10."""
+        # chars that are not allowed next to the param_id
+        bad_chars = string.ascii_letters + string.digits + '._'
+        id_len = len(param_id)
+        loc = 0
+        while True:
+            if loc >= len(prose):
+                return prose
+            next_loc = prose[loc:].find(param_id)
+            if next_loc < 0:
+                return prose
+            loc += next_loc
+            if loc > 0 and prose[loc - 1] in bad_chars:
+                loc += id_len
+                continue
+            end_loc = loc + id_len
+            if end_loc == len(prose) or prose[end_loc] not in bad_chars:
+                prose = prose[:loc] + param_str + prose[end_loc:]
+                loc += len(param_str)
+                continue
+            loc += id_len
```

Comment:
I believe this could be replaced with string replacement using regex

Reply:
I was having trouble with non-capturing groups in re.sub but I found a way to make it work nicely with regex.

Reply:
This is great, thanks